msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,322 @@
+### 4.3 App description
+
+The app descriptions in your manifest and store should be same and follow these guidelines:
+
+* Use "Microsoft Teams" on first reference. Subsequent references can be shortened to ”Teams”.
+* Do not abbreviate "Microsoft" (such as "MS" or "MSFT").
+* Avoid unnecessary capitalization and language errors.
+* Follow the [short description guidelines](~/concepts/deploy-and-publish/appsource/prepare/detail-page-checklist.md#short-description).
+* Follow the [long description guidelines](~/concepts/deploy-and-publish/appsource/prepare/detail-page-checklist.md#long-description).
+* Clearly describe any limitations, conditions, or exceptions to the functionality, features, and deliverables described in the listing and related materials before the customer acquires your offer. The capabilities you declare must relate to the core functions and description of your offer.
+
+### 4.4 Localization
+
+Your package must include a localization file that includes language translations that display based on the Teams language settings. The file must conform to the [Teams localization schema](~/concepts/build-and-test/apps-localization.md).
+
+## 5.0 Tab design
+
+See comprehensive [tab design guidelines](~/tabs/design/tabs.md).
+
+### 5.1 Setup
+
+Tab set up must not dead-end the new user. A way forward message should be provided to complete the workflow.
+
+Authentication should happen during tab setup and not after.
+
+Tabs must have a Help tab with instructions on how to do things such as sign up, configure, and use your app.
+
+### 5.2 Views
+
+* The available tab sign-in screen area use must not use large logos or display an entire webpage.
+* The tab view can be simplified by breaking them into multiple tabs to reduce the complexity.
+* The personal app should not have a duplicate header. Remove the logo from the iframe since the personal app framework already displays the app icon and app name.
+
+### 5.3 Navigation
+
+* Tabs must not have more than three levels of navigation.
+* The secondary and tertiary pages in a tab must be opened in a level two and level three view in the main tab area that is navigated via breadcrumbs or a left nav. You can also include these components to aid tab navigation:
+    * Back buttons
+    * Page headers
+    * Hamburger menus
+* Tab should not have horizontal scroll.
+* Deep links in tabs must not link to an external webpage but instead link somewhere within Teams (for example, task modules or other tabs).
 
-### &#9989; Tab content must not have excessive chrome or layered navigation
+### 5.4 Usability
 
-* Tabs must provide focused content and avoid needless UI elements. This usually refers to unnecessary nested or layered navigation, an extraneous or irrelevant UI next to the content, or any links that take the user to unrelated content. For example, the following tab view omits navigation menus and only showcases the main content:
+* Users must be able to undo their last action in the app.
```

Comment:
Just checking: Does this apply to all apps or just tabs?

Reply:
Applicable for Tab as it uses entire screen where the condition comes handy

Reply:
```suggestion
* Users must be able to undo their last action in the tab.
```